### PR TITLE
Migrate test_skipping.py from testdir to pytester

### PR DESCRIPTION
**Pull request by cmecklenborg**
Partial fix for #7942

Converts test_skipping.py from testdir to pytester. All tests are passing locally.

### CODE REVIEW

File: testing/test_skipping.py [hunk 1]
```diff
@@ -148,16 +150,16 @@ def test_func(self):
...
         skipped = evaluate_skip_marks(item)
         assert skipped
         assert skipped.reason == "condition: config._hackxyz"
 
 
 class TestXFail:
     @pytest.mark.parametrize("strict", [True, False])
-    def test_xfail_simple(self, testdir, strict):
-        item = testdir.getitem(
+    def test_xfail_simple(self, pytester, strict) -> None:
```
Comment (bluetech) [inline testing/test_skipping.py:161]: Would be great to catch the `pytester`s that appear in the middle of the parameter list, like here. A search/replace of ` pytest, ` with ` pytester: Pytester, ` should do the trick (also for the rest of the file). Also in this case case can annotate `strict`, like so:

```suggestion
    def test_xfail_simple(self, pytester: Pytester, strict: bool) -> None:
```

**Review by bluetech — COMMENTED** [review]
Thanks for working on this @cmecklenborg!

This looks good to me with one comment.

**Review by nicoddemus — APPROVED** [review]
Thanks @cmecklenborg!

I'm leaving this approved, but please take a look at @bluetech's comment/suggestion. 👍

File: testing/test_skipping.py (hunk 1 above)
Comment (cmecklenborg) [reply testing/test_skipping.py]: Argh, sorry for missing those! I added typehinting for the remaining `pytester` args, as well as for the `strict` args throughout.

**Pull request merged**